examples/classification_example/main_train.py

```diff
@@ -0,0 +1,238 @@
+
+    scale = np.sqrt(3.0 / embedding_dim)
+    table = np.empty(
+        [extractor.size(), embedding_dim], dtype=np.float32
+    )
+    oov = 0
+    for word, index in extractor.items():
+        if word in embed_dict:
+            embedding = embed_dict[word]
+        elif word.lower() in embed_dict:
+            embedding = embed_dict[word.lower()]
+        else:
+            embedding = np.random.uniform(
+                -scale, scale, [1, embedding_dim]
+            ).astype(np.float32)
+            oov += 1
+        table[index, :] = embedding
+    return torch.from_numpy(table)
+
+
+def create_model(schemes: Dict[str, Dict[str, BaseExtractor]],
+                 config: Config, in_channels: int):
+    text_extractor: BaseExtractor = schemes["text_tag"]["extractor"]
+    embedding_dict = {}
+    fake_tensor = torch.tensor([0.0 for i in range(100)])
+    for word, index in text_extractor.items():
+        embedding_dict[word] = fake_tensor
+
+    word_embedding_table = \
+        construct_word_embedding_table(embedding_dict, text_extractor)
+
+    model: nn.Module = \
+        CNN_Classifier(in_channels=in_channels,
+                       word_embedding_table=word_embedding_table)
+
+    if config.config_model.model == "bert":
+        model: nn.Module = BERTClassifier()
+
+    return model, word_embedding_table
+
+
+def train(model: nn.Module, optim: Optimizer, batch: Batch, max_sen_length: int):
+    word = batch["text_tag"]["data"]
+    labels = batch["label_tag"]["data"]
+    optim.zero_grad()
+
+    logits, pred = None, None
+
+    if config.config_model.model == "cnn":
+        logits, pred = model(batch)
+
+    if config.config_model.model == "bert":
+        logits, pred = model(pad_each_bach(word, max_sen_length))
+
+    labels_1D = torch.squeeze(labels)
+    torch.add(labels_1D, -2, out=labels_1D)
+
+    loss = criterion(logits, labels_1D)
+
+    loss.backward()
+    optim.step()
+
+    batch_train_err = loss.item() * batch.batch_size
+
+    return batch_train_err
+
+
+# All the configs
+config_data = yaml.safe_load(open("config_data.yml", "r"))
+config_model = yaml.safe_load(open("config_model.yml", "r"))
+
+config = Config({}, default_hparams=None)
+config.add_hparam('config_data', config_data)
+config.add_hparam('config_model', config_model)
+
+tp_request = {
+    "scope": Sentence,
+    "schemes": {
+        "text_tag": {
+            "entry_type": Token,
+            "attribute_get": "text",
+            "vocab_method": "indexing",
+            "type": TrainPreprocessor.DATA_INPUT,
+            "extractor": AttributeExtractor
+        },
+        "label_tag": {
+            "entry_type": Sentence,
+            "attribute_get": "speaker",
+            "vocab_method": "indexing",
+            "type": TrainPreprocessor.DATA_OUTPUT,
+            "extractor": AttributeExtractor
+        }
+    }
+}
+
+# All not specified dataset parameters are set by default in Texar.
```

Comment:
You can remove this comment as the explanation is included in documentation now.

Reply:
Removed